src/vm/vm_memory/memory_segments.rs

```diff
@@ -128,6 +133,54 @@ impl MemorySegmentManager {
             Err(MemoryError::WriteArg)
         }
     }
+
+    pub fn get_memory_holes(
+        &self,
+        accessed_addresses: &HashSet<Relocatable>,
+    ) -> Result<usize, MemoryError> {
+        let segment_used_sizes = self
+            .segment_used_sizes
+            .as_ref()
+            .ok_or(MemoryError::MissingSegmentUsedSizes)?;
+
+        let mut accessed_offsets_sets = HashMap::new();
+        for addr in accessed_addresses {
+            if addr.segment_index < 0 {
+                return Err(MemoryError::AddressInTemporarySegment(addr.segment_index));
+            }
```

Comment:
[It is](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/lang/vm/memory_segments.py#L197), but in assert form.
I just moved it so that no extra computations are executed if this condition is not met.